src/module.c

```diff
@@ -614,6 +615,10 @@ void moduleFreeContext(RedisModuleCtx *ctx) {
          * outside of call() context (timers, events, etc.). */
         if (--server.module_ctx_nesting == 0 && !server.core_propagates)
             propagatePendingCommands();
+        if (!server.module_ctx_nesting && server.busy_module) {
+            blockingOperationEnds();
+            server.busy_module = 0;
+        }
```

Comment:
please use one external `if (--server.module_ctx_nesting == 0)` and two internal `if`s